### PR TITLE
[Github] Improve automatic new issue labelling

### PR DESCRIPTION
**Pull request by cor3ntin**
This patch:
  * Replaces `andymckay/labeler` which does not appear to be maintained by github official solution
  * Removes the closed issue workflow which was disabled a few years ago and never fixed.
  * Adds a few rules to add label based on PR title, hopefully that can make triaging simpler. If that turns out to be useful, we can consider adding more rules for backends, etc. We could technically also pattern match the body of the issue but I'm concerned about trying to be _too_ clever.

The new system is only triggered on PR open so manual labels should not be removed.

### CODE REVIEW

**Comment by cor3ntin** [comment]
Full disclosure, this was not tested, not sure how testing could be done.

File: .github/workflows/new-issues.yml [hunk 1]
```diff
@@ -9,12 +9,14 @@ permissions:
...
-      pull-requests: write  # for andymckay/labeler to label PRs
+      issues: write
     runs-on: ubuntu-latest
     if: github.repository == 'llvm/llvm-project'
     steps:
-      - uses: andymckay/labeler@1.0.4
+      - uses: github/issue-labeler@v3.2
         with:
-          add-labels: 'new issue'
-          ignore-if-labeled: true
```
Comment (Endilll) [inline .github/workflows/new-issues.yml:20]: Do we understand how this labeler behaves in presence of manually added labels?
We definitely don't want to fight with our tools again.

File: .github/workflows/new-issues-labeler.yml
Comment (Endilll) [inline .github/workflows/new-issues-labeler.yml:1]: There are keywords to recognize backends (e.g. `AArch64`, `NVPTX`, `AMDGPU`) and LLVM components (e.g. `SelectionDAG`, `SelectionDAGISel`, `AsmParser`). I think they are worth listing here.

File: .github/workflows/new-issues.yml (hunk 1 above)
Comment (cor3ntin) [reply .github/workflows/new-issues.yml:20]: The workflow only triggers when PRs are created, not when edited.
But to be sure, I've set sync-labels to false explicitly https://github.com/github/issue-labeler#syncing-labels

**Review by Endilll — APPROVED** [review]

**Comment by cor3ntin** [comment]
@Endilll are you happy adding more rules in a separate commit?

**Pull request merged**